crates/polars-time/src/truncate.rs

```diff
@@ -59,7 +59,7 @@ impl PolarsTruncate for DateChunked {
                 if let Some(every) = every.get(0) {
                     let every = Duration::parse(every);
                     let w = Window::new(every, every, offset);
-                    self.try_apply(|t| {
+                    self.try_apply_values(|t| {
```

Comment:
this function looks expensive enough that it may be worth skipping null values for - with more consistent naming, this potential inefficiency jumps out a bit more